code/modules/crew_objectives/civilian_objectives.dm

```diff
@@ -54,16 +54,49 @@
 	else
 		return ..()
 
-/datum/objective/crew/responsibility
-	explanation_text = "Make sure nobody dies with alcohol poisoning."
+/datum/objective/crew/cocktail
+	explanation_text = "Have a bottle(any type) that contains 'something' when the shift ends. Each of them must be at least 'something'u."
 	jobs = "bartender"
+	var/targetchems = list()
+	var/chemnames = ""
+	var/chemsize
+	var/datum/reagent/chempath
 
-/datum/objective/crew/responsibility/check_completion()
-	for(var/mob/living/carbon/human/H in GLOB.mob_list)
-		if(H.stat == DEAD && H.drunkenness >= 80)
-			if((H.z in SSmapping.levels_by_trait(ZTRAIT_STATION)) || SSshuttle.emergency.shuttle_areas[get_area(H)])
-				return ..()
-	return TRUE
+/datum/objective/crew/cocktail/New()
+	. = ..()
+	var/mydrink = list()
+	for(var/i in 1 to 5)
+		chempath = get_random_reagent_id(CHEMICAL_GOAL_BARTENDER_SERVING)
+		if(!(chempath in targetchems))
+			targetchems += chempath
+			mydrink += "[initial(chempath.name)]"
+	for(var/i in 1 to length(mydrink)-1)
+		chemnames += "[mydrink[i]], "
+	chemnames += "and [mydrink[length(mydrink)]]"
+	qdel(mydrink)
+	chemsize = 4+(5-length(targetchems))
+	update_explanation_text()
```

Comment:
This may be impossible if 2 selected reagents have a reaction together

Reply:
`Q. Can reagents mix to each other?
A. When I took a look at them, there's no conflict.
I had to remove all basic drinks from the list not to let them mix each other. It's a shame that they're not the list, but it should be completed anyway.`

It's fine as long as there's no basic chems.